test/functional/wallet_decommission_request.py

```diff
@@ -0,0 +1,334 @@
+            self.log.debug(f'Tip: {tip_id}')
+
+            # Submit a valid transaction
+            output = {
+                    'Transfer': [ { 'Coin': 50_000 * ATOMS_PER_COIN }, { 'PublicKey': {'key': {'Secp256k1Schnorr' : {'pubkey_data': pub_key_bytes}}} } ],
+            }
+            encoded_tx, tx_id = make_tx([reward_input(tip_id)], [output], 0)
+            self.log.debug(f"Encoded transaction {tx_id}: {encoded_tx}")
+
+
+            self.setup_pool_and_transfer([encoded_tx])
+
+            # sync the wallet
+            assert_in("Success", await wallet.sync())
+
+            # check wallet best block if it is synced
+            best_block_height = await wallet.get_best_block_height()
+            assert_in(best_block_height, '1')
+
+            balance = await wallet.get_balance()
+            assert_in("Coins amount: 50000", balance)
+
+            assert_in("The transaction was submitted successfully", await wallet.create_stake_pool(40000, 0, 0.5, decommission_pub_key_encoded))
+            transactions = node.mempool_transactions()
+
+            self.gen_pos_block(transactions, 2)
+            assert_in("Success", await wallet.sync())
+
+            pools = await wallet.list_pool_ids()
+            assert_equal(len(pools), 1)
+            assert_equal(pools[0].balance, 40000)
+
+            # try decommission from hot wallet
+            assert_in("Controller error: Wallet error: Cannot use partially signed transaction in a decommission command",
+                       await wallet.decommission_stake_pool(pools[0].pool_id))
+
+            # create decommission request
+            decommission_req_output = await wallet.decommission_stake_pool_request(pools[0].pool_id)
+            decommission_req = decommission_req_output.split('\n')[1]
+
+            # try to sign decommission request from hot wallet
+            assert_in("Controller error: Wallet error: Input cannot be signed",
+                       await wallet.sign_raw_transaction(decommission_req))
+
+        decommission_signed_tx = ""
+
+        async with WalletCliController(node, self.config, self.log, chain_config_args=["--chain-pos-netupgrades", "true"]) as wallet:
+            # open cold wallet
+            await wallet.open_wallet("cold_wallet")
+            assert_in("Success", await wallet.sync())
+
+            # sign decommission request
+            decommission_signed_tx_output = await wallet.sign_raw_transaction(decommission_req)
+            decommission_signed_tx = decommission_signed_tx_output.split('\n')[1]
+
+        async with WalletCliController(node, self.config, self.log, chain_config_args=["--chain-pos-netupgrades", "true"]) as wallet:
+            # open hot wallet
+            await wallet.open_wallet("hot_wallet")
+
+            assert_in("The transaction was submitted successfully", await wallet.submit_transaction(decommission_signed_tx))
+
+            self.gen_pos_block(node.mempool_transactions(), 3)
```

Comment:
Can we add a check to see if that particular transaction is included in the mempool?